### PR TITLE
RSpec should ignore existing `RAILS_ENV`

### PR DESCRIPTION
**Pull request by c960657**
When running tests in the Docker development container, you need to manually specify `RAILS_ENV=test` to override [the default](https://github.com/mastodon/mastodon/blob/main/.devcontainer/docker-compose.yml#L11) specified in docker-compose.yml.

Running tests with `RAILS_ENV` set to `development` or `production` does not work, so there is no point in RSpec honoring the value already specified.

### CODE REVIEW

**Review by mjankowski — COMMENTED** [review]
The rspec generator default uses `||=` ...

https://github.com/rspec/rspec-rails/blob/v6.1.2/lib/generators/rspec/install/templates/spec/rails_helper.rb#L3

I agree that there's no real use case for running rspec w/ any non-`test` environment, though.

This change seems like another side effect of the boot.rb env check?... https://github.com/mastodon/mastodon/blob/main/config/boot.rb#L3-L13

**Comment by c960657** [comment]
@mjankowski I am not sure what you are asking?

With `||=`, the [predefined value](https://github.com/mastodon/mastodon/blob/main/.devcontainer/compose.yaml#L10) of `RAILS_ENV=development` is not overwritten.

If we remove `RAILS_ENV` from compose.yaml, you would have to specify `RAILS_ENV` every time you run e.g. `rails console`. We don't want that either.

**Comment by mjankowski** [comment]
I don't know if I was asking anything ... just observing that the need for even having `RAILS_ENV` in the compose yaml is also a side effect of requiring the env var in that boot check.

All that said - having done a whole bunch of "let's make this less broken and more in line with doing vaguely what you expect it to" sort of work on the various devcontainer/Dockerfile realm a few weeks ago -- I think I agree with this change. As long as the var is required by the boot script and set in [that compose yaml](https://github.com/mastodon/mastodon/blob/6952af137d17f65e65963e4d9ba129060fad9d3d/.devcontainer/compose.yaml#L10) it is indeed both required to and annoying to need to explicitly set the rails env to run tests ... when that's quite obviously what you intend to happen there.

I hereby endorse this change with the full weight of my office (this is not substantial).

**Review by mjankowski — APPROVED** [review]

**Comment by c960657** [comment]
This is an old PR, but I still think it is a good idea, and I hope it will be considered for merge.

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-ENV['RAILS_ENV'] ||= 'test'
+ENV['RAILS_ENV'] = 'test'
```
Comment (mjankowski) [inline spec/rails_helper.rb:3]: 📛📛📛

Comment (c960657) [reply spec/rails_helper.rb:3]: What does that emoji mean? :-)

Comment (mjankowski) [reply spec/rails_helper.rb:3]: I've now triple-badge-verified your change.

_(But note again, this has zero weight or impact)_

Comment (mjankowski) [reply spec/rails_helper.rb:3]: Would you be willing to receive a quadruple badge if I can create one ... or do you think three is sufficient?

Comment (c960657) [reply spec/rails_helper.rb:3]: Three is enough, I think. But an animated GIF would be nice.

**Pull request closed without merging**